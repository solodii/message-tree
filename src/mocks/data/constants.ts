export const NONEXISTENT_MESSAGE = 'NONEXISTENT_MESSAGE';
export const NONEXISTENT_SUBJECT = 'NONEXISTENT_SUBJECT';
