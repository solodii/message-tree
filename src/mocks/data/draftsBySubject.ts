import {
  State,
  emptySubject
} from '../../reducers/draftsBySubject';

const data: State = {
  [emptySubject]: {
    subjectId: emptySubject,
    text: ''
  },
  '6c6f1976-ff5b-420b-bc9d-f7484f0ee53f': {
    subjectId: '6c6f1976-ff5b-420b-bc9d-f7484f0ee53f',
    text: 'fourth'
  }
};

export default data;

export const validSubject = '6c6f1976-ff5b-420b-bc9d-f7484f0ee53f';
export const invalidSubject = '';
export const nonexistentSubject = 'nonexistent-subject';
