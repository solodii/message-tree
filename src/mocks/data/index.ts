export * from './constants';
export {
  drafts,
  draftsBySubject,
  invalidDraft,
  validDraft
} from './drafts';
export {
  messages,
  messagesById,
  messagesBySubject,
  messagesWithDraft,
  messagesWithoutDraft,
  messageTrees
} from './messages';
export { state } from './state';
