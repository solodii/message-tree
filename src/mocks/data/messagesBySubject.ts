import {
  State,
  emptySubject
} from '../../reducers/messagesBySubject';

const data: State = {
  [emptySubject]: [
    {
      subjectId: emptySubject,
      text: 'first',
      id: '0a6a7bfd-a212-4ae1-8125-6c7486213a94',
      date: new Date()
    },
    {
      subjectId: emptySubject,
      text: 'third',
      id: '913d97b3-6e04-4c1e-a11c-222221abf00e',
      date: new Date()
    }
  ],
  '0a6a7bfd-a212-4ae1-8125-6c7486213a94': [
    {
      subjectId: '0a6a7bfd-a212-4ae1-8125-6c7486213a94',
      text: 'second',
      id: '6c6f1976-ff5b-420b-bc9d-f7484f0ee53f',
      date: new Date()
    }
  ]
};

export default data;

export const nonexistentSubject = 'nonexistent-subject';
