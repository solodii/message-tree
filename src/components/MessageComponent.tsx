import * as React from 'react';
import * as dateformat from 'dateformat';
import { Message } from '../reducers/messagesBySubject';

export interface Props {
  message: Readonly<Message>;
}

export default function MessageComponent(props: Props): JSX.Element {
  const {
    message: {
      date,
      text
    }
  } = props;  
  return (
    <div className="message">
      <div><span>{dateformat(date)}</span></div>
      <div><span>{text}</span></div>
    </div>
  );
}
