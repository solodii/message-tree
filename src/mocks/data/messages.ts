import {
  Message,
  MessagesById,
  MessagesBySubject,
  MessageTree
} from '../../state/types';
import { ROOT_SUBJECT } from '../../state/constants';

export const messages: Message[] = [
  {
    subjectId: ROOT_SUBJECT,
    text: 'first',
    id: '0a6a7bfd-a212-4ae1-8125-6c7486213a94',
    date: new Date(2016, 10, 10, 10, 1)
  },
  {
    subjectId: ROOT_SUBJECT,
    text: 'second',
    id: '6c6f1976-ff5b-420b-bc9d-f7484f0ee53f',
    date: new Date(2016, 10, 10, 10, 2)
  },
  {
    subjectId: '0a6a7bfd-a212-4ae1-8125-6c7486213a94',
    text: 'third',
    id: '913d97b3-6e04-4c1e-a11c-222221abf00e',
    date: new Date(2016, 10, 10, 10, 3)
  }
];

export const messagesById: MessagesById = {
  [messages[0].id]: messages[0],
  [messages[1].id]: messages[1],
  [messages[2].id]: messages[2]
};

export const messagesBySubject: MessagesBySubject = {
  [messages[0].subjectId]: [
    messages[0],
    messages[1]
  ],
  [messages[2].subjectId]: [
    messages[2]
  ]
};

export const messagesWithDraft = [
  messages[0]
];

export const messagesWithoutDraft = [
  messages[1],
  messages[2]
];

export const messageTrees: MessageTree[] = [
  {
    message: messages[0],
    hasDraft: messagesWithDraft.indexOf(messages[0]) !== -1,
    children: [
      {
        message: messages[2],
        hasDraft: messagesWithDraft.indexOf(messages[2]) !== -1,
        children: []
      }
    ]
  },
  {
    message: messages[1],
    hasDraft: messagesWithDraft.indexOf(messages[1]) !== -1,
    children: []
  }
];
