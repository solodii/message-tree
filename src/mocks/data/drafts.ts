import {
  Draft,
  DraftsBySubject
} from '../../state/types';
import { ROOT_SUBJECT } from '../../state/constants';
import { messagesWithDraft } from './messages';

export const drafts: Draft[] = [
  {
    subjectId: ROOT_SUBJECT,
    text: ''
  }
];

messagesWithDraft.forEach((msg) => {
  drafts.push({
    subjectId: msg.id,
    text: `draft for ${msg.id}`
  });
});

export const draftsBySubject: DraftsBySubject = {};

drafts.forEach((draft) => {
  draftsBySubject[draft.subjectId] = draft;
});

export const invalidDraft = drafts[0];
export const validDraft = drafts[1];
