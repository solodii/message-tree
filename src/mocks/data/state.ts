import { State } from '../../state/types';
import { draftsBySubject } from './drafts';
import {
  messagesById,
  messagesBySubject
} from './messages';

export const state: State = {
  draftsBySubject,
  messagesById,
  messagesBySubject
};
