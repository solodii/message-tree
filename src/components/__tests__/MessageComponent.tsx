import * as React from 'react';
import { shallow } from 'enzyme';
import { messages } from '../../mocks/data';
import MessageComponent, { Props } from '../MessageComponent';

describe('MessageComponent', () => {
  const initialProps: Props = {
    message: messages[0]
  };
  const wrapper = shallow<Props, {}>(
    <MessageComponent {...initialProps} />
  );

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
