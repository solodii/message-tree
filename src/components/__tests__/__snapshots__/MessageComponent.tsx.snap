// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageComponent renders correctly 1`] = `
<div
  className="message"
>
  <div>
    <span>
      Thu Nov 10 2016 10:01:00
    </span>
  </div>
  <div>
    <span>
      first
    </span>
  </div>
</div>
`;
